Bonsai.Editor/Layout/LayoutHelper.cs

```diff
@@ -193,8 +193,13 @@ static IReadOnlyList<VisualizerFactory> GetMashupArguments(InspectBuilder builde
             if (visualizerMappings.Count == 0) return Array.Empty<VisualizerFactory>();
             return visualizerMappings.Select(mapping =>
             {
-                var nestedSources = GetMashupArguments(mapping.Source, typeVisualizerMap);
+                // mapping.Source == builder if two visualizers in the mashup visualize the same node
+                var nestedSources = mapping.Source == builder
+                    ? throw new WorkflowBuildException("Visualizer mappings of same node are combined together.", builder)
+                    : GetMashupArguments(mapping.Source, typeVisualizerMap);
+
```

Comment:
```suggestion
                // stack overflow happens if a visualizer ends up being mapped to itself
                if (mapping.Source == builder)
                    throw new WorkflowBuildException("Combining together visualizer mappings from the same node is not currently supported.", builder);
                
                var nestedSources = GetMashupArguments(mapping.Source, typeVisualizerMap);
```

For future refactoring and emphasis I would separate the detection into its own section. You are right about getting the right wording, I am not entirely sure myself how to make this less confusing, and ultimately there might not even be a good way to do this.

I ended up converging on something that ends with "not currently supported." as a way of hinting that this is something weird that we don't know whether it makes sense or not. The reason it might be helpful to hint this is so people reach out to us with a potentially really useful case that might motivate us to resolve this better.